src/App.tsx

```diff
@@ -3,16 +3,42 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+  movies: Movie[];
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+    movies: [...moviesFromServer],
+  };
+
+  handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = event.target;
+    const { query } = this.state;
+
+    this.setState({
+      query: value.toLowerCase(),
+
+      movies: [...moviesFromServer].filter(movie => (
```

Comment:
```suggestion
      movies: moviesFromServer.filter(movie => (
```
filter creates new array by default